ServantPersistent/src/Domain/Tenant.hs

```diff
@@ -0,0 +1,44 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Domain.Tenant
+    where
+
+import Control.Lens
+import Data.Time
+import Database.Persist
+import Control.Monad.IO.Class
+import Data.Text (Text)
+import Models
+import Types
+import Updater
+import DBTypes
+import Operation
+
+dbCreateTenant :: TenantInput -> App (Maybe TenantID)
+dbCreateTenant ti = runDb $ do
+    time <- liftIO $ getCurrentTime
+    let dbt = DBTenant { _dBTenantName = ti ^. name
+                       , _dBTenantBackofficeDomain = ti ^. backofficeDomain
+                       , _dBTenantOwnerId = Nothing
+                       , _dBTenantStatus = NewT
+                       , _dBTenantCreatedAt = time
+                       , _dBTenantUpdatedAt = time
+                       }
+    insertUnique dbt
+
+
+dbGetTenant :: TenantID -> App (Maybe Tenant)
+dbGetTenant = runDb . get
+
+dbGetTenantByName :: Text -> App (Maybe TenantID)
+dbGetTenantByName t = runDb $ do
+    ti <- getBy (UniqueTenant t)
```

Comment:
Yes, I did not realise that. I'll change that to the backoffice domain or the tenant id.